### PR TITLE
Align PortableTimer's behavior with System.Threading.Timer

### LINKED ISSUE #7: PeriodicBatchingSink writes first message immediately on .NET Core.
When using `PeriodicBatchingSink` on .NET Core first call to `Emit` causes immediate call to `EmitBatchAsync` with a single event passed in.

Works as expected on full framework.

### PR DESCRIPTION
**Pull request by skomis-mm**
fixes [#7](https://github.com/serilog/serilog-sinks-periodicbatching/issues/7)
also should cover [#4](https://github.com/serilog/serilog-sinks-periodicbatching/issues/4)

### CODE REVIEW

File: src/Serilog.Sinks.PeriodicBatching/Sinks/PeriodicBatching/PortableTimer.cs [hunk 1]
```diff
@@ -109,5 +115,13 @@ public void Dispose()
...
         }
     }
+
+    static class TaskExtensions
+    {
+        public static async void AsObserved(this Task task)
+        {
+            await task.ConfigureAwait(false);
+        } 
+    }
```
Comment (skomis-mm) [inline src/Serilog.Sinks.PeriodicBatching/Sinks/PeriodicBatching/PortableTimer.cs:125]: This retains fail fast behavior if `_onTick` handler throws

File: test/Serilog.Sinks.PeriodicBatching.Tests/Sinks/PeriodicBatching/PortableTimerTests.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
+﻿#if NETCOREAPP1_0
```
Comment (nblumhardt) [inline test/Serilog.Sinks.PeriodicBatching.Tests/Sinks/PeriodicBatching/PortableTimerTests.cs]: `#if !WAITABLE_TIMER` will keep this aligned with the way `PortableTimer` is conditionally included in the project being tested.

**Comment by nblumhardt** [comment]
Thanks! Sorry about the delay, it took me a little while to wrap my head around the change. Only a minor nitpick, otherwise looks good-to-go.

**Comment by skomis-mm** [comment]
Fixed, thanks!

**Pull request merged**